rails_programming/advanced_forms_and_activerecord/active_record_callbacks.md

```diff
@@ -88,3 +88,13 @@ Callbacks are useful and many, like `:after_create` and `:before_destroy`, are p
 This section contains helpful links to other content. It isn't required, so consider it supplemental.
 
 * [WikiBooks Reference on Callbacks](http://en.wikibooks.org/wiki/Ruby_on_Rails/ActiveRecord/Callbacks)
+
+### Knowledge Check
+This section contains questions for you to check your understanding of this lesson. If you're having trouble answering the questions below on your own, review the material above to find the answer.
+
+ * <a class='knowledge-check-link' href='https://guides.rubyonrails.org/active_record_callbacks.html#callback-registration'>How do you register a callback?</a>
+ * <a class='knowledge-check-link' href='https://guides.rubyonrails.org/active_record_callbacks.html#relational-callbacks'>How can you implement a user's images being destroyed if said user is destroyed?</a>
+  * <a class='knowledge-check-link' href='https://guides.rubyonrails.org/active_record_callbacks.html#using-if-and-unless-with-a-symbol'>What's the difference between using a symbol or a Proc for a conditional callback?</a>
```

Comment:
The main thing we'll want to get users thinking about is how they can make callbacks run only under certain conditions. Focusing on how they work under the hood is getting a little too much into the weeds imo. Would you mind changing this one to something like this instead?
> How can you make a callback run only when certain conditions are met?

Which would link to: https://www.theodinproject.com/lessons/ruby-on-rails-active-record-callbacks#specifying-callback-characteristics